daml-lf/repl/src/main/scala/com/digitalasset/daml/lf/repl/testing/Main.scala

```diff
@@ -419,13 +418,15 @@ object Repl {
           case Some(DValue(_, _, body, _)) =>
             val expr = argExprs.foldLeft(body)((e, arg) => EApp(e, arg))
 
+            val compiledPackages = PureCompiledPackages(state.packages).right.get
             val machine =
-              Speedy.Machine.fromExpr(
-                expr = expr,
-                compiledPackages = PureCompiledPackages(state.packages).right.get,
-                scenario = false,
+              Speedy.Machine(
+                compiledPackages = compiledPackages,
                 submissionTime = Time.Timestamp.now(),
                 initialSeeding = InitialSeeding.NoSeed,
+                expr = SEApp(compiledPackages.compiler.unsafeCompile(expr), Array(SEValue.Token)),
```

Comment:
Why do we have a token here now? That doesn’t look right.

Reply:
Good catch